Client/Frontend/Browser/Playlist/Controllers/PlaylistListViewController.swift

```diff
@@ -664,6 +667,73 @@ extension PlaylistListViewController {
       animated: false,
       scrollPosition: .none)
   }
+  
+  func loadSharedPlaylist(folderSharingUrl: String) {
+    sharedFolderLoadingTask?.cancel()
+    
+    sharedFolderLoadingTask = Task { @MainActor in
+      // Shared Folder already exists
+      if let existingFolder = PlaylistFolder.getSharedFolder(sharedFolderUrl: folderSharingUrl) {
+        PlaylistManager.shared.currentFolder = existingFolder
+        self.loadingState = .fullyLoaded
+        return
+      }
+      
+      // Shared Folder doesn't exist
+      do {
+        self.playerView.setStaticImage(image: UIImage())
+        let model = try await PlaylistSharedFolderNetwork.fetchPlaylist(folderUrl: folderSharingUrl)
+        let folder = await PlaylistSharedFolderNetwork.createInMemoryStorage(for: model)
+        PlaylistManager.shared.currentFolder = folder
+        self.loadingState = .partial
+        
+        if let folderImageUrl = model.folderImage {
+          Task { @MainActor in
```

Comment:
`@MainActor` here is redundant, you are already inside of a `MainActor` isolated `Task`

Reply:
Fixed.